File: 3o Período/Aeds-II/TP01/Q02-PalindromoC/palindromo.c
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

#define MAXTAM 100

bool ehFim(char input[]){
    bool resp = false;
    if(input[0] == 'F' && input[1] == 'I' && input[2] == 'M') 
        resp = true;
    return resp;
}//fim ehFim()

bool ehPalindromo(char input[]) {
    int tam = strlen(input) - 1;
    bool resp = true;

    for (int i = 0; i < tam / 2; i++) {
        if (input[i] != input[tam - 1 - i])
            resp = false;
    }

    return resp;
}

int main(){
    char input[MAXTAM] = {0};

    fgets(input, sizeof(input), stdin);

    while(!ehFim(input)){
        bool resp = ehPalindromo(input); 
        if(resp) printf("SIM\n");
        else printf("NAO\n");

        fgets(input, sizeof(input), stdin);
    }
    return 0;
}//fim main

File: 3o Período/Aeds-II/TP01/Q01-PalindromoJava/Palindromo.java
import java.util.Scanner;

public class Palindromo{
    /*
    * Metodo publico para avaliar se e' fim do input
    * @params String input
    * @returns boolean resp
    */
    static public boolean ehFim(String input){
        boolean resp = false;
        if(input.charAt(0) == 'F' && input.charAt(1) == 'I' && input.charAt(2) == 'M')
            resp = true;
        return resp;
    }// fim ehFim

    /*
    * Metodo que analisa a string e retorna se e' um palindromo
    * @params String str
    * @returns boolean resp
    */
    static public boolean ehPalindromo(String str){
        boolean resp = true;
        int tam = str.length();

        for(int i = 0; i < tam/2; i++){
            if(str.charAt(i) != str.charAt(tam - 1 - i)) resp = false;
        }

        return resp;
    }//fim ehPalindromo


    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        while (!ehFim(input)) {
            boolean resp = ehPalindromo(input);
            if(resp) System.out.println("SIM");
            else System.out.println("NAO");

            input = scanner.nextLine();
        } 

        scanner.close();

    }// fim main
}